### PR TITLE
Move to addon v2 blueprint

### PR DESCRIPTION
**Pull request by mkszepp**
The addon was already a v2 addon, but the addon v2 blueprint was missing, because it was one of the first v2 addon and blueprint was not ready

With this PR we work with v2 addon conventions. Its easier to handle update... in the future.

This is the final step to get ready for next release

Which this change, the drop of node < 18 is not anymore braking, so i think, we can also ship as a minor update all previous changes

### CODE REVIEW

File: ember-moment/babel.config.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "plugins": [
+    "@embroider/addon-dev/template-colocation-plugin",
+    ["babel-plugin-ember-template-compilation", {
+      "targetFormat": "hbs",
+      "transforms": []
+    }],
+    ["module:decorator-transforms", { "runtime": { "import": "decorator-transforms/runtime" } }],
```
Comment (NullVoxPopuli) [inline ember-moment/babel.config.json:8]: Does ember-moment use decorators?

If so, using this plugin has requirements of the consuming project (ember-cli-babel 8.2+)

**Review by NullVoxPopuli — APPROVED** [review]

File: ember-moment/babel.config.json (hunk 1 above)
Comment (mkszepp) [reply ember-moment/babel.config.json:8]: looked into code... no there are not decorator... but i have discovered seen, that we have still `classic classes` in this addon 😮 (see service)

Comment (mkszepp) [reply ember-moment/babel.config.json]: `decorator-transforms` is now removed. thank you for hint

Comment (NullVoxPopuli) [reply ember-moment/babel.config.json]: Excellent, thank you!

**Pull request merged**